src/bci_build/package/gcc/README.md.j2

```diff
@@ -0,0 +1,95 @@
+
+The GNU Compiler Collection (GCC) is an optimizing compiler for various
+architectures and operating systems. It is the default compiler of the GNU
+project and the default compiler for most Linux Distributions including SUSE
+Linux Enterprise and openSUSE.
+
+
+## How to use this image
+
+### Compile your application in a `Dockerfile`
+
+Generally you'd want to compile your application and distribute it as part of a
+container image of your own. For that, create a `Dockerfile` as follows which
+builds a container image based on this image, copies the sources to a working
+directory and compiles the application:
+{%- set reference_no_release = (image.reference | replace("-%RELEASE%", "") ) %}
+
+```Dockerfile
+FROM {{ reference_no_release }}
+WORKDIR /src/
+COPY . /src/
+RUN gcc main.c src1.c src2.c
+CMD ["./a.out"]
+```
+
+It is also possible to compile a static binary with gcc as part of a multistage
+build:
+
+```Dockerfile
+FROM {{ reference_no_release }} as builder
+WORKDIR /src/
+COPY . /src/
+RUN gcc -static main.c src1.c src2.c
+
+FROM registry.suse.com/bci/bci-micro:latest
+WORKDIR /build/
+COPY --from=builder /src/a.out /build/app
+CMD ["/build/app"]
+```
+
+Note that you must build a static binary to deploy it into bci-micro as
+otherwise shared libraries might be missing. You cannot deploy such an app into
+a `scratch` image as it is not possible to statically link glibc.
+
+
+### Available build systems
+
+The Container Image comes preloaded with `make` by default. Other build systems
+and related utilities are available in the repository and can be installed via
+`zypper`, including:
+- `meson`
+- `cmake`
+- `ninja`
+- `autoconf` & `automake`
+
+
+### Available compiler frontends
+
+The GNU Compiler Collections supports a multitude of frontends. The container
+image ships the C and C++ frontends available as `gcc` and `g++`
+respectively. The following additional frontends can be installed from the
+repository:
+- `gcc{{ image.version }}-fortran` for Fortran support
+{% if (image.version == 7) or (image.is_opensuse) %}- `gcc{{ image.version }}-ada` for the Ada frontend (GNAT)
+{% endif %}{% if image.is_opensuse %}- `gcc{{ image.version }}-go` for the Go frontend
+- `gcc{{ image.version }}-objc` and `gcc{{ image.version }}-obj-c++` for the Objective C and Objective C++
+{% if (image.version >= 12) %}- `gcc{{ image.version }}-d` for the frontend to the D Language
+{% endif %}{%- if (image.version >= 13) %}- `gcc{{ image.version }}-m2` for the Modula 2 compiler frontend
+{% endif %}{%- if (image.version >= 14) %}- `gcc{{ image.version }}-rust` for the GNU Rust compiler
+{% endif %}{% endif %}
+
+### Using the container image interactively
+
+You can also utilize this container image in a one-shot fashion where building a
+full container image is not desirable. The easiest way is to mount the working
+directory of your application into the launched container and compile your
+application there:
```

Comment:
```suggestion
You can use the image to create ephemeral containers that execute only gcc. This can be useful in situations, where building a
full container image is not practical. One way to do this is to mount the working directory of an application into the launched container and compile the
application there:
```